### DISCUSSION TITLE
Kral Arthur kale güvercin sorusu

### DISCUSSION THREAD

**Discussion by sonsuzus** [sonsuzus/sonsuzus#29] [Soru Cevap]
Kral Arthur’un ülkesi toplam n tane kaleden oluşmaktadır ve kaleler arasında iletişim için güvencinler kullanılmaktadır. Bir eğitimli güvercin eğitim aldığı kale ikilisinin birinden bırakıldığında diğerine uçmaktadır. Her eğitimli güvercin her gün eğitim aldığı kaleler arasında git gel yapmaktadır. Her bir kale ikilisi arasında doğrudan güvercinlerle iletişim kurulabilmesi için n*(n-1)/2 güvercin eğitilmesi gerekmektedir. Güvercin eğitimi masraflı olduğundan Kral Arthur her bir kale ikilisi için güvercin eğitilmemesini, aralarında doğrudan iletişim için güvercin eğitilmemiş kalelerin diğer kaleler üzerinden iletişimlerini sağlamalarını istemiştir. Örneğin; (A,B), (B, C) ve (C, D) kale ikilileri için güvercinler eğitilmiş ise, A ve D kaleleri, B ve C kaleleri aracılığı ile iletişim sağlayabilir. Ayrıca Kral Arthur eğitimli güvercinlerden herhangi biri öldüğünde dahi bütün kalelerin birbirleriyle iletişim kurabilmelerini istemektedir.

Kral Arthur’un ülkesinde toplam 121 kale varsa, Kral Arthur’un istediği şartları sağlamak için en az kaç güvercin eğitilmesi gerekmektedir?

**Comment by yigidi** [comment]
Verilen durumda, Kral Arthur'un istediği şartları sağlamak için kaç güvercinin en az eğitilmesi gerektiğini hesaplamak istiyoruz. Toplam 121 kale olduğunu biliyoruz. Her bir kale ikilisi arasında doğrudan güvercinlerle iletişim kurabilmesi için n*(n-1)/2 güvercin gerekiyor.

121 kale için bu formülü kullanarak hesaplayalım:

n*(n-1)/2 = 121

n*(n-1) = 242

n^2 - n - 242 = 0

Bu ikinci dereceden denklemin pozitif çözümü n = 16'dır. Ancak, her ikilinin iletişim kurması gerektiği unutulmamalıdır. Bu nedenle, 16 kaleden oluşan bir grup olduğunda, her ikilinin iletişim kurabilmesini sağlamak için 120 güvercin gerekecektir.

Ancak Kral Arthur, eğitimli güvercinlerden herhangi birinin ölmesi durumunda bile iletişimin devam etmesini istiyor. Bu nedenle, her bir iletişim rotası için 2 güvercin (bir yön için bir güvercin, diğer yön için bir güvercin) yedek olarak düşünülebilir.

Bu durumda, 120 güvercinin yerine her bir iletişim rotası için 240 güvercin eğitilmiş gibi düşünebiliriz.

Sonuç olarak, Kral Arthur'un istediği şartları sağlamak için en az 240 güvercin eğitilmesi gerekmektedir.
